### PR TITLE
Change substring to comply with standard sql definition

### PR DESCRIPTION
**Pull request by saurabhd336**
TPCH's query 24 https://github.com/apache/pinot/blob/master/pinot-integration-tests/src/test/resources/tpch/24.sql fails due to pinot's `substring` definition being different from standard sql definition. As per https://www.postgresql.org/docs/15/functions-string.html

```
substring ( string text [ FROM start integer ] [ FOR count integer ] ) → text

Extracts the substring of string starting at the start'th character if that is specified, and stopping after count characters if that is specified. Provide at least one of start and count.

substring('Thomas' from 2 for 3) → hom

substring('Thomas' from 3) → omas

substring('Thomas' for 2) → Th
```

This change fixes the tpch query, but introduces a backward-incopatible change. Will wait for reviews.

### CODE REVIEW

**Comment by KKcorps** [comment]
I remember discussing this for Tableau as well. The reason it wasn't done is cause it breaks backward compatibility for people already using this function.

**Comment by xiangfu0** [comment]
If this is new syntax, then it won't break the backward compatibility right?

**Review by xiangfu0 — APPROVED** [review]

File: pinot-common/src/main/java/org/apache/pinot/common/function/scalar/StringFunctions.java [hunk 1]
```diff
@@ -79,29 +79,27 @@ public static String upper(String input) {
   /**
    * @see String#substring(int)
    * @param input Parent string
-   * @param beginIndex index from which substring should be created
+   * @param beginIndex 1 based index from which substring should be created
    * @return substring from beginIndex to end of the parent string
    */
   @ScalarFunction(names = {"substr", "substring"})
```
Comment (xiangfu0) [inline pinot-common/src/main/java/org/apache/pinot/common/function/scalar/StringFunctions.java:85]: Shall we consider keep `substr` as current behavior and use `substring` for postgresql behavior?

**Comment by saurabhd336** [comment]
@xiangfu0 sure let's go with having both substr and substring as separate functions

**Comment by xiangfu0** [comment]
Do we support syntax like: `substring ( string text [ FROM start integer ] [ FOR count integer ] )` in calcite?

**Comment by saurabhd336** [comment]
@xiangfu0 yes calcite sql parser itself parses `substring ( string text [ FROM start integer ] [ FOR count integer ] )` into the correct substring function node

**Comment by xiangfu0** [comment]
> @xiangfu0 yes calcite sql parser itself parses `substring ( string text [ FROM start integer ] [ FOR count integer ] )` into the correct substring function node

Nice, then please add a test for that as well.

**Comment by saurabhd336** [comment]
@xiangfu0 , tpch's query 24 already tests that https://github.com/apache/pinot/blob/master/pinot-integration-tests/src/test/resources/tpch/24.sql hence i didn't add an explicit one.

**Pull request merged**